include/camp/resource.hpp

```diff
@@ -172,18 +172,28 @@ namespace resources
 #endif
 
   template<typename Res>
-  struct EventProxy {
+  struct EventProxy : ::camp::resources::detail::EventProxyBase {
+    using native_event = typename std::decay<decltype(std::declval<Res>().get_event())>::type;
+
     EventProxy(EventProxy &&) = default;
-    EventProxy(EventProxy const &) = delete;
+    EventProxy(EventProxy &) = delete;
     EventProxy &operator=(EventProxy &&) = default;
-    EventProxy &operator=(EventProxy const &) = delete;
+    EventProxy &operator=(EventProxy &) = delete;
 
     EventProxy(Res r) :
       resource_{r}
     {}
 
-    Event get() {
-      return resource_.get_event_erased();
+    native_event get() {
+      return resource_.get_event();
+    }
+
+    template <typename T = Res,
+              typename = typename std::enable_if<
+                  !std::is_same<typename std::decay<decltype(std::declval<T>().get_event())>::type,
+                                Event>::value>::type>
+    operator native_event() {
+      return resource_.get_event();
     }
 
     operator Event() {
```

Comment:
SFINAE to allow EventProxy<Resource> to work. @trws what do you think of this fix?

Reply:
I like it, may want to factor this out into something more easily identifiable but I love the result.